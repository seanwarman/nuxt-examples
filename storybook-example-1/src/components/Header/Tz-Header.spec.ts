import { mount, VueWrapper } from '@vue/test-utils'
import TzHeader from './Tz-Header.vue'
import { expect, describe, it, beforeEach } from 'vitest'

let wrapper: VueWrapper

describe('SiteHeader.vue setup', () => {
    beforeEach(() => {
        wrapper = mount(TzHeader, {
            props: {},
            slots: {},
        })
    })

    it('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it('emits a logo clicked event', async () => {
        const element = wrapper.find('header div:first-child')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick() // wait until $emits have been handled
        expect(wrapper.emitted().menuClicked).toBeTruthy()
    })

    it('emits a logo clicked event', async () => {
        const element = wrapper.find('header div:nth-child(2)')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick() // wait until $emits have been handled
        expect(wrapper.emitted().logoClicked).toBeTruthy()
    })

    it('emits a wishlist clicked event', async () => {
        const element = wrapper.find('header div:last-child div:first-child')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick() // wait until $emits have been handled
        expect(wrapper.emitted().wishlistClicked).toBeTruthy()
    })

    it('emits a basket clicked event', async () => {
        const element = wrapper.find('header div:last-child div:nth-child(2)')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick() // wait until $emits have been handled
        expect(wrapper.emitted().basketClicked).toBeTruthy()
    })

    it('emits a my account clicked event', async () => {
        const element = wrapper.find('header div:last-child div:nth-child(3)')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick() // wait until $emits have been handled
        expect(wrapper.emitted().myAccountClicked).toBeTruthy()
    })

    it('emits a search updated event when search updated', async () => {
        const element = wrapper.find('button')
        element.trigger('click') //simulate click
        await wrapper.vm.$nextTick()
        expect(wrapper.emitted().searchClicked).toBeTruthy()
    })
})
