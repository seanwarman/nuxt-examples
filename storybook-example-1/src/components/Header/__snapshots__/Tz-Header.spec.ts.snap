// Vitest Snapshot v1

exports[`SiteHeader.vue setup > renders correctly 1`] = `
<header
  class="grid grid-cols-12 gap-2 p-[10px] sm:px-[20px] lg:px-[40px] xl:px-[80px]"
>
  <div
    class="order-1 col-span-4 md:hidden"
  >
    <svg
      class="w-5"
      fill="none"
      viewBox="0 0 32 32"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3 5a2 2 0 0 1 2-2h22a2 2 0 1 1 0 4H5a2 2 0 0 1-2-2Zm0 11a2 2 0 0 1 2-2h12a2 2 0 1 1 0 4H5a2 2 0 0 1-2-2Zm0 11a2 2 0 0 1 2-2h22a2 2 0 1 1 0 4H5a2 2 0 0 1-2-2Z"
        fill="currentColor"
        fill-rule="evenodd"
      />
    </svg>
  </div>
  <div
    class="order-2 col-span-4 flex justify-center md:order-1 md:col-span-3 md:justify-start xl:col-span-1"
  >
    <img
      alt="tredz logo, the online bike store for cycle enthusiasts"
      class="h-7"
      src="/src/assets/logos/svgs/master-logo.svg"
      title="tredz.co.uk, the online bike store for cycle enthusiasts"
    />
  </div>
  <div
    class="tr-site-header__search order-4 col-span-12 md:order-2 md:col-span-6"
  >
    <button
      class="flex w-full items-center justify-start gap-2 rounded-full bg-gray-100 py-1 px-3 text-gray-400"
    >
      <svg
        class="w-5 text-gray-500"
        fill="none"
        viewBox="0 0 32 32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M30.414 30.414a2 2 0 0 1-2.828 0L17.672 20.5l2.828-2.828 9.914 9.914a2 2 0 0 1 0 2.828Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M25 13c0 6.627-5.373 12-12 12S1 19.627 1 13 6.373 1 13 1s12 5.373 12 12Zm-12 8a8 8 0 1 0 0-16 8 8 0 0 0 0 16Z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
       Search 
    </button>
  </div>
  <div
    class="order-3 col-span-4 flex justify-end gap-2 md:col-span-3 xl:col-span-5"
  >
    <div
      class="tr-site-header__control"
    >
      <span
        class="flex w-[30px] items-center justify-center rounded-full bg-gray-100 p-[4px] hover:bg-gray-200"
      >
        <svg
          fill="none"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6.03 7.539c-.83.657-1.53 1.813-1.53 3.988 0 .835.369 1.952 1.149 3.308.763 1.328 1.837 2.743 3.048 4.137 2.42 2.784 5.265 5.34 6.912 6.75.153.131.405.143.594-.046l3.236-3.237a1.5 1.5 0 0 1 2.122 2.122l-3.237 3.236c-1.25 1.25-3.273 1.396-4.666.204-1.69-1.447-4.663-4.114-7.225-7.061-1.28-1.473-2.489-3.05-3.385-4.61-.88-1.53-1.548-3.194-1.548-4.803 0-2.893.976-5 2.666-6.34C5.793 3.9 7.86 3.5 9.73 3.5c2.179 0 4.123 1.041 5.497 2.506.257.275.499.568.72.877 1.315-1.965 3.44-3.383 6.019-3.383 1.814 0 3.949.303 5.66 1.534 1.805 1.298 2.875 3.43 2.875 6.493 0 1.567-.773 3.197-1.718 4.655-.975 1.504-1.282 2.047-2.646 3.485a1.5 1.5 0 1 1-2.176-2.065c1.301-1.372 1.47-1.764 2.305-3.052.865-1.335 1.235-2.353 1.235-3.023 0-2.303-.759-3.434-1.626-4.058-.962-.691-2.345-.969-3.91-.969-2.368 0-4.464 2.377-4.464 5.027a1.5 1.5 0 0 1-3 0c0-1.212-.542-2.488-1.462-3.469C12.12 7.08 10.93 6.5 9.729 6.5c-1.508 0-2.806.331-3.7 1.039Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="tr-site-header__control"
    >
      <span
        class="flex w-[30px] items-center justify-center rounded-full bg-gray-100 p-[4px] hover:bg-gray-200"
      >
        <svg
          fill="none"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M6 10h20a4 4 0 0 1 4 4v13a4 4 0 0 1-4 4H6a4 4 0 0 1-4-4V14a4 4 0 0 1 4-4Zm0 3a1 1 0 0 0-1 1v13a1 1 0 0 0 1 1h20a1 1 0 0 0 1-1V14a1 1 0 0 0-1-1H6Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M23 8v7a7 7 0 1 1-14 0V8a7 7 0 1 1 14 0Zm-7-4a4 4 0 0 0-4 4v7a4 4 0 0 0 8 0V8a4 4 0 0 0-4-4Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      class="tr-site-header__control"
    >
      <span
        class="flex w-[30px] items-center justify-center rounded-full bg-gray-100 p-[4px] hover:bg-gray-200"
      >
        <svg
          fill="none"
          viewBox="0 0 32 32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16 13a4 4 0 1 0 0-8 4 4 0 0 0 0 8Zm0 3a7 7 0 1 0 0-14 7 7 0 0 0 0 14Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
          <path
            clip-rule="evenodd"
            d="M24.621 24.83c-.286-1.685-1.142-3.253-2.459-4.47C20.523 18.846 18.306 18 16 18s-4.523.846-6.162 2.36c-1.317 1.217-2.173 2.785-2.46 4.47.274.18.612.38 1.018.585C9.874 26.161 12.333 27 16 27c3.667 0 6.126-.84 7.604-1.585.406-.205.744-.405 1.017-.585ZM28 26.087c0-2.94-1.264-5.76-3.515-7.84C22.235 16.168 19.183 15 16 15s-6.235 1.168-8.485 3.247C5.265 20.327 4 23.147 4 26.087 4 26.087 7.515 30 16 30s12-3.913 12-3.913Zm-2.487-1.95Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </div>
  </div>
</header>
`;
